### PR TITLE
updated codebase so that images can be relative to the HTML file

### PR DESCRIPTION
**Pull request by bashaus**
I was having some issues with running posthtml under gulp as relative image files were not being identified relative to the HTML file.

I removed a lot of the ES6 functionality and upgraded some of the other files

### CODE REVIEW

File: lib/imgAutosize.js [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+            const promises = [];
+
+            tree.match({ tag: 'img' }, img => {
+                img.attrs = img.attrs || {};
+
+                const imagePath = translatePath(projectRoot, htmlRelativePath, img.attrs.src);
+
+                const updateWidth =
+                    img.attrs.width == 'auto' ? true :
+                    !img.attrs.width ? options.processEmptySize : false;
```
Comment (maltsev) [inline lib/imgAutosize.js]: A double ternary operator is hard to understand. Can you replace it with something more readable (e.g. multiple ifs)?

File: lib/imgAutosize.js [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+                const updateHeight =
+                    img.attrs.height == 'auto' ? true :
+                    !img.attrs.height ? options.processEmptySize : false;
```
Comment (maltsev) [inline lib/imgAutosize.js]: A double ternary operator is hard to understand. Can you replace it with something more readable (e.g. multiple ifs)?

File: test/imgAutosize.js [hunk 3]
```diff
@@ -37,8 +35,8 @@ describe('Plugin', () => {
 
         it('should autosize remote PNG', () => {
             return init(
-                '<img src="http://dummyimage.com/111x52/000/fff.png" width="auto" height="25" alt="hi">',
-                '<img src="http://dummyimage.com/111x52/000/fff.png" width="111" height="25" alt="hi">'
+                '<img src="http://placehold.it/200x200" width="auto" height="25" alt="hi">',
+                '<img src="http://placehold.it/200x200" width="200" height="25" alt="hi">'
```
Comment (maltsev) [inline test/imgAutosize.js:39]: I guess it should be `width="25"` because the image is a square and height is 25 px.

**Review by maltsev — CHANGES_REQUESTED** [review]
Thanks for the PR!

I think we could drop Node 0.12 support as it's [no longer maintained](https://github.com/nodejs/Release) and [the tests fail](https://travis-ci.org/posthtml/posthtml-img-autosize/jobs/345789813). Can you replace it with Node 4 in `.travis.yml`?

File: test/imgAutosize.js (hunk 3 above)
Comment (bashaus) [reply test/imgAutosize.js:39]: This was just based on the existing code that was there - should I make a new issue for this?

**Comment by bashaus** [comment]
Hi there

Resolved support for Node version 4 and still removed the requirement for bluebird - I've also had to expand util.promisify as that only appeared in version 8.

In terms of the other code change you suggested, this is based on the existing unit test. If you think this should be changed, can we open a separate issue for this?

**Comment by bashaus** [comment]
Updated dependencies and setup version 0.1.2, hope you don't mind :)

File: test/imgAutosize.js (hunk 3 above)
Comment (maltsev) [reply test/imgAutosize.js:39]: Sorry, I just misread the param values. You're right, the bug was there already before your changes.

> should I make a new issue for this?

Yes, please.

**Pull request merged**